compiler/lib/driver.ml

```diff
@@ -98,28 +98,34 @@ let ( +> ) f g x = g (f x)
 let map_fst f (x, y, z) = f x, y, z
 
 let effects ~deadcode_sentinal p =
-  if Config.Flag.effects ()
-  then (
-    if debug () then Format.eprintf "Effects...@.";
-    let p, live_vars = Deadcode.f p in
-    let p = Effects.remove_empty_blocks ~live_vars p in
-    let p, live_vars = Deadcode.f p in
-    let info = Global_flow.f ~fast:false p in
-    let p, live_vars =
-      if Config.Flag.globaldeadcode ()
-      then
-        let p = Global_deadcode.f p ~deadcode_sentinal info in
-        Deadcode.f p
-      else p, live_vars
-    in
-    p |> Effects.f ~flow_info:info ~live_vars +> map_fst Lambda_lifting.f)
-  else
-    ( p
-    , (Code.Var.Set.empty : Effects.trampolined_calls)
-    , (Code.Var.Set.empty : Effects.in_cps) )
+  match Config.effects () with
+  | Some _ ->
+      if debug () then Format.eprintf "Effects...@.";
+      let p, live_vars = Deadcode.f p in
+      let p = Effects.remove_empty_blocks ~live_vars p in
+      let p, live_vars = Deadcode.f p in
+      let info = Global_flow.f ~fast:false p in
+      let p, live_vars =
+        if Config.Flag.globaldeadcode ()
+        then
+          let p = Global_deadcode.f p ~deadcode_sentinal info in
+          Deadcode.f p
+        else p, live_vars
+      in
+      p
+      |> Effects.f ~flow_info:info ~live_vars
+      |> map_fst
+           (match Config.effects () with
```

Comment:
You can match the inner of the option directly, just name the backend on line 102.